flecsi: loosen cxxstd for boost, kokkos and hpx

rbberger opened the pull request:
only prevents older standards than C++17

rbberger committed:
flecsi: loosen cxxstd for boost, kokkos and hpx

only prevents older standards than C++17

opensdh commented on repos/spack_repo/builtin/packages/flecsi/package.py:
```diff
@@ -125,6 +125,11 @@ class Flecsi(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("^hpx networking=none", when="backend=hpx")
 
+    for cxxstd in ("11", "14"):
+        conflicts(f"^kokkos cxxstd={cxxstd}")
```
Can `conflicts("^kokkos cxxstd=11")` be a problem if (some version of) Kokkos doesn't have that variant at all, or is that considered irrelevant because it's `conflicts`?

rbberger replied:
It could be a problem, yes. I just realized that kokkos has a hard conflict for its own `cxxstd=11` and `cxxstd=14` for any version newer or equal to `4.0.0`. So that one is actually redundant. @lucbv FYI, you can probably get rid of the variant value cxxstd=11 and remove the conflict altogether in the kokkos recipe.

rbberger committed:
flecsi: simplify kokkos dependency

rbberger committed:
e4s: restore ^boost cxxstd=17 requirement

Previously forced by flecsi. Removing the restriction caused the concretizer to
sometimes use c++26, which isn't supported by the selected compilers in these
stacks. As a workaround, restore the restriction for these CI stacks.

rbberger commented:
@eugeneswalker needed to make a change for the CI to not do something different. More details in the commit message.
